openssl-dynamic/src/main/c/ssl.c

```diff
@@ -678,21 +935,34 @@ TCN_IMPLEMENT_CALL(jint, SSL, getError)(TCN_STDARGS,
     return SSL_get_error(ssl_, ret);
 }
 
-// How much did SSL write into this BIO?
-TCN_IMPLEMENT_CALL(jint /* nbytes */, SSL, pendingWrittenBytesInBIO)(TCN_STDARGS,
-                                                                     jlong bio /* BIO * */) {
-    BIO *b = J2P(bio, BIO *);
+TCN_IMPLEMENT_CALL(jint /* nbytes */, SSL, bioCtrlPending)(TCN_STDARGS, jlong bio /* BIO* */) {
+    BIO* b = J2P(bio, BIO*);
 
     if (b == NULL) {
         tcn_ThrowException(e, "bio is null");
         return 0;
     }
 
-    UNREFERENCED_STDARGS;
-
     return BIO_ctrl_pending(b);
 }
 
+#ifdef OPENSSL_IS_BORINGSSL
```

Comment:
IIRC I was trying to be a "good citizen" when writing BIO and use existing APIs when possible but its possible I'm actually abusing these interfaces.

Here is how they are described in the header files:
https://github.com/openssl/openssl/blob/master/include/openssl/bio.h#L90
```c
# define BIO_CTRL_WPENDING       13/* opt - number of bytes still to write */
```

https://github.com/openssl/openssl/blob/master/include/openssl/bio.h#L87
```c
# define BIO_CTRL_PENDING        10/* opt - is their more data buffered */
```

I'm using `BIO_CTRL_PENDING` to mean "how many bytes are pending in the current ByteBuffer" and I'm using `BIO_CTRL_WPENDING` to mean "how many bytes are pending in the `char[]` which is part of the BIO" (which must be written to the peer). I could just create another method(s) which don't hook into the `BIO_*ctrl` mechanism if that is safer/more clear ... @davidben WDYT?